azuredevops/internal/service/core/resource_team.go

```diff
@@ -0,0 +1,558 @@
+package core
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ahmetb/go-linq"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/core"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/identity"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	securityhelper "github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+func ResourceTeam() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceTeamCreate,
+		Read:     resourceTeamRead,
+		Update:   resourceTeamUpdate,
+		Delete:   resourceTeamDelete,
+		Importer: tfhelper.ImportProjectQualifiedResourceUUID(),
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"administrators": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotWhiteSpace,
+				},
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Set:        schema.HashString,
+			},
+			"members": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotWhiteSpace,
+				},
+				Optional:   true,
+				Computed:   true,
```

Comment:
The members are read anyway, so why not having the ability to process them at a later place inside a HCL?

Reply:
If member is configured in HCL,  the property here is not computed.  https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#computed

Reply:
If you read the description there's the following at the end `...and can be used as output for interpolation into other resources`. If you don't mark a property computed it cannot be used somewhere else. If you have a look in the AzureRM Terraform provider the tuple of `Optional: true` and `Computed: true` is all over the place. 

For example at the `Key Vault` resource for the `access_policy` property:

![image](https://user-images.githubusercontent.com/14177833/120225793-305f8e00-c246-11eb-8250-64474c718ab1.png)

Reply:
Key vault will add at least one policy by default (user who created  it).  If the property value not set, service will auto set a default value for it, then this property can be `computed`.  Otherwise if we don't do anything and service won't do anything too, this property should not be an `computed` value.

Reply:
No, the Key Vault resource does not create any access policies automatically:

![image](https://user-images.githubusercontent.com/14177833/120309041-0dc88600-c2d5-11eb-9669-d9b05a8aa219.png)

![image](https://user-images.githubusercontent.com/14177833/120309073-14ef9400-c2d5-11eb-9b1d-cb1d7b9cec2b.png)

That's because of the fact that there is a [`azurerm_key_vault_access_policy`](azurerm_key_vault_access_policy) to create access policies aside the key vault resource.

And if you create an access policy using the `azurerm_key_vault_access_policy`

![image](https://user-images.githubusercontent.com/14177833/120313432-3b63fe00-c2da-11eb-8e94-fcb2b1dcc812.png)

You'll find this policy in the `key vault` resource because the `access_policy`attribute is `computed`

![image](https://user-images.githubusercontent.com/14177833/120313500-59316300-c2da-11eb-9ec7-ff5d8c8fa2ed.png)

And thus you can use the `access_policy` from the `key_vault` resource in other calculations.

In the end I followed this strategy while I implemented the `azuredevops_teams` resource.

Reply:
1. Currently the `accessPolicies`  is a required parameter:
   ```log
    {
        "error": {
            "code": "BadRequest",
            "message": "The parameter accessPolicies is not specified."
        }
    }
    ```
    I don‘t ’ know which API version used when key vault was introduced to AzureRM, from the AzureRM schema design that the `accessPolicies` used to be an optional but now it is required. AzureRM still works is because the request will auto set the `accessPolicies` to an empty array if `access_policy` not configured. [expandAccessPolicies](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/keyvault/access_policy_schema.go#L175)
   
2. All configurations in `azurerm_key_vault` are in one API request,  key vault/access policies can be created or update with one request.

3. Back to ADO teams,   `admin` and `member` is not part of the team API,  you need call another APIs to manage admin or members, there is three different API here.   Combine these APIs as one resource is a good idea, treat them like one API.  But I still think there is not strong reason here that we should add `Computed: true` for `member`, `members` areismanaged by explicit configuration and API calls.

Reply:
@xuzhang3 how can we reach an agreement here. I'd like to get this PR merged and it seems people are waiting for it actually.